src/coreclr/src/vm/vars.hpp

```diff
@@ -694,7 +694,7 @@ typedef DPTR(GSCookie) PTR_GSCookie;
 #define READONLY_ATTR
 #else
 #ifdef __APPLE__
-#define READONLY_ATTR_ARGS section("__TEXT,__const")
+#define READONLY_ATTR_ARGS section("__DATA,__const")
```

Comment:
Did you mean to change it only for `CORECLR_EMBEDDED`? See the previous discussion: https://github.com/dotnet/runtime/pull/36808#discussion_r428583329.

Reply:
@am11 do you see a reason to make this different for embedded and regular coreclr? It seems to me that it is fine to have it in the __DATA,__const segment  for both.

Reply:
If it is working for all cases, then sounds good. :)
I was just thinking that since we have `__TEXT,__const	_s_gsCookie` in the osx binary from the beginning (and previous change did not alter the section type); considering the requirements such as Mojave hardened runtime, maybe we do not need to change the type for the existing (non-embedded) case.

Reply:
I've verified that it works with the non-embedded coreclr with Mojave hardened runtime (running the app signed and from bundle) too.